dashboard/test/controllers/scripts_controller_test.rb

```diff
@@ -471,6 +471,25 @@ class ScriptsControllerTest < ActionController::TestCase
     assert script.hidden
   end
 
+  test 'cannot update if changes have been made to the database which are not reflected in the current edit page' do
+    sign_in @levelbuilder
+    Rails.application.config.stubs(:levelbuilder_mode).returns true
+
+    script = create :script
+    stub_file_writes(script.name)
+
+    error = assert_raises RuntimeError do
+      post :update, params: {
+        id: script.id,
+        script: {name: script.name},
+        script_text: '',
+        old_script_text: 'different'
+      }
+    end
+
+    assert_includes error.message, 'Could not update the script because the contents of one of its lessons or levels has changed outside of this editor. Reload the page and try saving again.'
+  end
```

Comment:
good test! I think it would be a good idea to have one test verifying that the request succeeds when old_script_text does match for a script with at least one lesson and one level, by passing in `old_script_text: ScriptDSL.serialize_lesson_groups(script)`.